src/main/java/org/datadog/jmxfetch/util/CustomLogger.java

```diff
@@ -1,123 +1,166 @@
+                return String.format(format,
+                    simpleDateFormat.format(new Date()).toString(),
+                    // NOTE(remy): these conversions may generate a lot of garbage
+                    //      (jaime): if performance impact over a sustained period of time
+                    //               isn't too bad, we can probably live with this.
+                    LogLevel.fromJulLevel(lr.getLevel()).toString(),
+                    simpleClassName(lr.getSourceClassName()),
+                    lr.getMessage()
+                );
+            }
+        };
+
+        // prepare the different handlers
+        // --
+
+        ConsoleHandler stdoutHandler = null;
+        ConsoleHandler stderrHandler = null;
+        FileHandler fileHandler = null;
+
+        // the logLocation isn't always containing a file, it is sometimes
+        // referring to a standard output. We want to create a FileHandler only
+        // if the logLocation is a file on the FS.
+        if (logLocation != null && logLocation.length() > 0) {
+            if (!isStdOut(logLocation) && !isStdErr(logLocation)) {
+                // file logging
+                try {
+                    // maximum one 5MB file
+                    fileHandler = new FileHandler(logLocation, 5 * 1024 * 1024, 1);
+                    fileHandler.setFormatter(formatter);
+                } catch (Exception e) {
+                    fileHandler = null;
+                    log.error("can't open the file handler");
+                }
+            } else if (isStdErr(logLocation)) {
+                // console handler sending on stderr
+                // note that ConsoleHandler is sending on System.err
+                stderrHandler = new ConsoleHandler();
+                stderrHandler.setFormatter(formatter);
+                stderrHandler.setLevel(level.toJulLevel());
             }
         }
 
-        // replace default appender with the correct layout
-        LoggerConfig loggerConfig = config.getLoggerConfig(LogManager.ROOT_LOGGER_NAME);
-        loggerConfig.removeAppender(target);
+        // always have a console handler sending the logs to stdout
+        stdoutHandler = new StdoutConsoleHandler();
+        stdoutHandler.setFormatter(formatter);
+        stdoutHandler.setLevel(level.toJulLevel());
 
-        Appender appender = ConsoleAppender.newBuilder()
-                    .setConfiguration(config)
-                    .withName(target)
-                    .withLayout(layout)
-                    .build();
-        appender.start();
+        // should configure the root logger
+        // ---
 
-        loggerConfig.addAppender(appender, null, null);
-        loggerConfig.setLevel(level);
+        Logger logger = Logger.getLogger("");
 
-        ctx.updateLoggers();
+        // clean all existing handlers
+        for (Handler handler : logger.getHandlers()) {
+            logger.removeHandler(handler);
```

Comment:
I'm guessing we want to flush the handlers before removing. They also may need closing.

Reply:
definitely was the intended change 🤦 fixing that, good catch 🙇

Reply:
we don't really mind this handler, but that's not harming anyone so I'll close them first 👍